addons/l10n_in_edi_ewaybill/models/account_move.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # Transaction Details
+    l10n_in_transaction_type = fields.Selection(
+        [
+            ("1", "Regular"),
+            ("2", "Bill To-Ship To"),
+            ("3", "Bill From-Dispatch From"),
+            ("4", "Combination of 2 and 3"),
+        ],
+        string="Transaction Type (E-waybill)",
+        tracking=True,
+    )
+    l10n_in_type_id = fields.Many2one("l10n.in.ewaybill.type", "Document Type", tracking=True)
+    l10n_in_subtype_id = fields.Many2one("l10n.in.ewaybill.type", "Sub Supply Type", tracking=True)
+
+    # transportation details
+    l10n_in_distance = fields.Integer("Distnace", tracking=True)
+    l10n_in_mode = fields.Selection([
+        ("0", "Managed by Transporter"),
+        ("1", "By Road"),
+        ("2", "Rail"),
+        ("3", "Air"),
+        ("4", "Ship")],
+        compute="_compute_l10n_in_mode", store=True, readonly=False,
+        string="Transportation Mode", copy=False, tracking=True)
+
+    # Vehicle Number and Type required when transportation mode is By Road.
+    l10n_in_vehicle_no = fields.Char("Vehicle Number", copy=False, tracking=True)
+    l10n_in_vehicle_type = fields.Selection([
+        ("R", "Regular"),
+        ("O", "ODC")],
+        string="Vehicle Type", copy=False, tracking=True)
+
+    # Document number and date required in case of transportation mode is Rail, Air or Ship.
+    l10n_in_transportation_doc_no = fields.Char(
+        string="Document Number",
+        help="""Transport document number. If it is more than 15 chars, last 15 chars may be entered""",
```

Comment:
should not we just do the trimming when eneratin the json then?

Reply:
we don't trim it but if the Transport documents have more than 15 then in odoo only enter the last 15 chars.